src/coreclr/jit/morph.cpp

```diff
@@ -2805,17 +2805,8 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
         noway_assert(arg != nullptr);
         call->gtCallCookie = nullptr;
 
-#if defined(TARGET_X86)
-        // x86 passes the cookie on the stack as the final argument to the call.
-        GenTreeCall::Use** insertionPoint = &call->gtCallArgs;
-        for (; *insertionPoint != nullptr; insertionPoint = &((*insertionPoint)->NextRef()))
-        {
-        }
-        *insertionPoint = gtNewCallArgs(arg);
-#else  // !defined(TARGET_X86)
-        // All other architectures pass the cookie in a register.
+        // All architectures pass the cookie in a register.
         call->gtCallArgs = gtPrependNewCallArg(arg, call->gtCallArgs);
-#endif // defined(TARGET_X86)
```

Comment:
What is the reason for this change? Did you measure impact of it? Since `ebx` is callee saved this might be pretty expensive to use as the new register given how few callee-saved registers are available on x86.

Reply:
calli using this helper is slow path for multiple difference reasons. We are optimizing for simplicity here. This change is deleting number of instructions from the helper that I expect will more than compensate for the ebx use here.

To make this go fast, we would need to get rid of this helper completely and call the marshaling stub directly from here. It would be more involved change, beyond the scope of this PR.

Reply:
My main worry was that reserving ebx can bias the register allocator into spilling more variables that are live across calls which will affect even the paths that do not run the expensive calli. However I checked win x86 diffs and there is only a single non-test diff in libraries with this change (https://github.com/dotnet/runtime/blob/cffaa78235ea93d5e3eeb56956579df503e11250/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs#L144-L146) so it seems fine to me.